### DISCUSSION TITLE
Is it possible to have nested tokens?

### DISCUSSION THREAD

**Discussion by joapuiib** [pygments/pygments#2793] [Q&A] [answered]
Hello,
I'm trying to build a custom lexer that highlights a relational database model with a specific syntax.
- Primary keys are red and underlined (`pk` class).
- Foreign keys are italic (`fk` class).
- Entity names are blue (`entity` class)

I'm migrating from [highlightjs](https://github.com/highlightjs/highlight.js), where I was able to achieve it:

![image](https://github.com/user-attachments/assets/6a355f66-4af4-4793-8521-cb73c7236d22)

```html
<span class="fk">
<span class="pk">#id_cicle</span>(<span class="entity">CICLE</span>:<span class="pk">#id_cicle</span>)
</span>
````

In the previous image, a Foreign Key definition contains Primary Keys and an entity name.
In that case, we have a `<span class="fk">` that wraps other `<span>` with other highlight classes.

__In Pygments, is it possible to have nested tokens?__

I've been looking at some Pygments lexer implementations, but I couldn't find anything similar.

Thanks in advance!

**Comment by birkenfeld** [comment, marked as answer]
Unfortunately, such a feature is not implemented.

**Reply by joapuiib** [reply]
I'm looking for alternatives to achieve the same output.

Would it be possible to append two tokens to the same text?

```html
<span class="fk pk">#id_cicle</span>
<span class="fk">(</span>
<span class="fk entity">CICLE</span>
<span class="fk">:</span>
<span class="fk pk">#id_cicle</span>
<span class="fk">)</span>
```
> Note that `<span class="fk pk">#id_cicle</span>` would have two tokens, Primary Key and Foreign Key.

**Reply by birkenfeld** [reply]
No, each bit of text can only be assigned one token type.

**Reply by joapuiib** [reply]
Understood :+1: 

Thanks for your answers!
